runtime/src/tiered_storage/footer.rs

```diff
@@ -0,0 +1,331 @@
+
+// The ending 8 bytes of a valid tiered account storage file.
+pub const FOOTER_MAGIC_NUMBER: u64 = 0x502A2AB5; // SOLALABS -> SOLANA LABS
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[repr(C)]
+pub struct TieredStorageMagicNumber(pub u64);
+
+impl Default for TieredStorageMagicNumber {
+    fn default() -> Self {
+        Self(FOOTER_MAGIC_NUMBER)
+    }
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountMetaFormat {
+    #[default]
+    Hot = 0,
+    Cold = 1,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountDataBlockFormat {
+    #[default]
+    AlignedRaw = 0,
+    Lz4 = 1,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum OwnersBlockFormat {
+    #[default]
+    LocalIndex = 0,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountIndexFormat {
+    // This format does not support any fast lookup.
+    // Any query from account hash to account meta requires linear search.
+    #[default]
+    Linear = 0,
+    // Similar to index, but this format also stores the offset of each account
+    // meta in the index block.
+    LinearIndex = 1,
+}
+
+#[derive(Debug)]
+pub struct TieredFileFormat {
+    pub meta_entry_size: usize,
+    pub account_meta_format: AccountMetaFormat,
+    pub owners_block_format: OwnersBlockFormat,
+    pub account_index_format: AccountIndexFormat,
+    pub data_block_format: AccountDataBlockFormat,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[repr(C)]
+pub struct TieredStorageFooter {
+    // formats
+    pub account_meta_format: AccountMetaFormat,
+    pub owners_block_format: OwnersBlockFormat,
+    pub account_index_format: AccountIndexFormat,
+    pub data_block_format: AccountDataBlockFormat,
+
+    // account-related
+    pub account_meta_count: u32,
+    pub account_meta_entry_size: u32,
+    pub account_data_block_size: u64,
+
+    // owner-related
+    pub owner_count: u32,
+    pub owner_entry_size: u32,
+
+    // offsets
+    // The offset of account meta+data blocks is omitted as it's always 0.
+    pub account_pubkeys_offset: u64,
+    pub owners_offset: u64,
```

Comment:
@brooksprumo: Only the account_pubkeys (index) offset and owners' offset are stored here.  Meaning the account meta and data are already stored together.